src/sfctl/state.py

```diff
@@ -0,0 +1,70 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# -----------------------------------------------------------------------------
+
+"""Read and modify state related to the CLI"""
+
+import os
+from datetime import datetime, timezone
+import jsonpickle
+from knack.config import CLIConfig
+from pkg_resources import get_distribution
+
+# knack CLIConfig has been re-purposed to handle state instead.
+
+# Default names
+SF_CLI_NAME = 'sfctl'
+SF_CLI_STATE_DIR = os.path.join('~', '.{0}'.format(SF_CLI_NAME))
+STATE_FILE_NAME = 'state'
+
+# Format: Year, month, day, hour, minute, second, microsecond, timezone
+DATETIME_FORMAT = "Year %Y Month %m Day %d %H:%M:%S:%f %Z"
+
+
+def get_state_path():
+
+    return CLIConfig(SF_CLI_STATE_DIR, SF_CLI_NAME, 'state').config_path
```

Comment:
No need to do all this. This is what it's calling into:

self.config_path = os.path.join(self.config_dir, CLIConfig._CONFIG_FILE_NAME)